src/QsCompiler/DocumentationParser/DocItem.cs

```diff
@@ -39,13 +45,25 @@ internal abstract class DocItem
         /// <param name="itemName">The name of the item itself</param>
         /// <param name="kind">The item's kind: operation, function, or UDT</param>
         /// <param name="documentation">The source documentation for the item</param>
-        internal DocItem(string nsName, string itemName, string kind, ImmutableArray<string> documentation)
+        internal DocItem(string nsName, string itemName, string kind, ImmutableArray<string> documentation,
+            IEnumerable<QsDeclarationAttribute> attributes)
         {
             namespaceName = nsName;
             name = itemName;
             uid = (namespaceName + "." + name).ToLowerInvariant();
             itemType = kind;
-            comments = new DocComment(documentation);
+            var depAttrs = attributes.Where(Microsoft.Quantum.QsCompiler.BuiltIn.MarksDeprecation).ToList();
```

Comment:
You could instead go with SymbolResolution.TryFindRedirect - that would take care of both determining whether the type or callable is deprecated and finding the redirect if one has been defined. It seems good to not duplicate that logic.

Reply:
Are we expecting the argument to @Deprecated to be the name of the replacement item, or a message? I've been assuming that it's a message, since there might not always be a simple replacement item (either there might not be any or there might be more than one).

Reply:
No, it's supposed to be the replacement item. Throughout our libraries that works well. If the string is empty, then there isn't a replacement item, if it's not, then it's assumed to be the full name of the replacement.

Reply:
OK, I think it's correct.